### ISSUE TITLE
Enhancement: Add a [consider-using-in]-checker

### ISSUE THREAD

**Issue by pheanex** [PyCQA/pylint#1977]
**I suggest adding a checker that detects boolOps where manual comparisons of a variable are made against multiple elements with "or"s instead of using "in" against a set of the elements combined**

Example:
```python
# bad
name == 'a' or name == 'b' or name == 'c'

# good
name in ('a', 'b', 'c')
```

It's in the same spirit of the existing RefactoringChecker ["consider-merging-isinstance"](https://github.com/PyCQA/pylint/blob/master/pylint/checkers/refactoring.py#L473).

If there are no objections, I'd add the test-cases here for discussion and then start implementing a visit_boolop check.

**Comment by PCManticore** [comment]
Sounds good! I suggest recommending using a set literal instead of a tuple, `name in {a, b, c}`.

**Comment by pheanex** [comment]
The problem with using a set literal as container is that it can not take unhashable types.
```python
>>> value = [1,2,3]
>>> value == [1,2,3] or value == [1,2]
True
>>> value in {[1, 2, 3], [1, 2]}
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: unhashable type: 'list'
>>> value in ([1, 2, 3], [1, 2])
True
```

For a first set of testcases I came up with the following:
```python
# bad
value = 1
a_set = {1, 2, 3}
a_list= [1, 2, 3]
a_str = '1'

value == 1 or value == 2 or value == 3  # [consider-using-in]
value == 1 or value == 1  # [consider-using-in]
value == 1 or value == 2  # [consider-using-in]
value == '1' or value == '2'  # [consider-using-in]
1 == value or 2 == value  # [consider-using-in]
1 == value or value == 2  # [consider-using-in]
value == a_set or value == a_list or value == a_str  # [consider-using-in]

value == 1  # value not checked against multiple values
value1 == 1 or value2 == 2  # different variables
value == 1 and value == 2  # not 'or'
value is 1 or value is 2  # 'in' compares using '==' not 'is'
value == 1 and value == 2 or value == 3  # not only 'or's
value == 1 or value == 2 or value == 3 and value == 4  # only simple cases considered for now
```

Should we also suggest the fix in the message?

**Comment by PCManticore** [comment]
That's a good observation @pheanex ! The test cases look good to me and I appreciate that we keep it just for simple cases, otherwise it might turn into a painful check to implement correctly. Suggesting the fix sounds good as well if it doesn't complicate the check too much,.
